tests/unit/Makefile

```diff
@@ -54,17 +54,30 @@ $(UNIT_TESTS)::
 	./$@
 endif
 
+$(VALGRIND_PEDANTIC_TESTS)::
```

Comment:
I am not really following. Why not just add ` --show-leak-kinds=all --errors-for-leak-kinds=all` to line 70? Are there too many errors to suppress if we do that?

Reply:
Different compiler implementations and libcrypto invoke different code paths and have different underlying implementations which leak memory. This PR only fixes and adds the pedantic check for Openssl 1.1.1 + GCC9. Adding the remaining configuration is a future task.

You can see the other ones fail in this [CI run](https://us-west-2.codebuild.aws.amazon.com/project/eyJlbmNyeXB0ZWREYXRhIjoidUpNb2QzNjZzQ3lrWm1UQ2xzdFZMN1ZGYWE5dVlSQ1ZKd1BnZzZ6aXN1dlZXTmFuZWdvNnR0SitPaFlIcUdjaDZ6RU9KemY2Mi9ybXVBajF0RXo4NmZ1Rnl4dkZrT241UUY4MTVQMkRuNTQ9IiwiaXZQYXJhbWV0ZXJTcGVjIjoiemFRUDgvZUMreVhER0RKaiIsIm1hdGVyaWFsU2V0U2VyaWFsIjoxfQ%3D%3D/batch/aa58120d-cb84-473a-853a-b2f55b09afb9)

Reply:
I can't actually see the other ones that fail, because the logs are currently deleting really quickly. I'm just trying to get a feel for how big the PR would be if we turn on these flags for all the compilers.

Reply:
Its not only a large PR but also the effort of testing and fixing for all those compilers. With this PR we incrementally enable pedantic testing and dont block on fixing everything.